transmission/writer.go

```diff
@@ -0,0 +1,77 @@
+package transmission
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sync"
+	"time"
+)
+
+// WriterOutput implements the Output interface by marshalling events to JSON
+// and writing to STDOUT, or to the writer W if one is specified.
+type WriterOutput struct {
+	W io.Writer
+
+	BlockOnResponses  bool
+	ResponseQueueSize uint
+	responses         chan Response
+
+	sync.Mutex
+}
+
+func (w *WriterOutput) Start() error {
+	w.responses = make(chan Response, w.ResponseQueueSize)
+	return nil
+}
+
+func (w *WriterOutput) Stop() error { return nil }
+
+func (w *WriterOutput) Add(ev *Event) {
+	var m []byte
+	func() {
```

Comment:
wtf I have no idea why.

Reply:
ah. In the old version, there was a lock that was taken at the beginning of the closure and a `defer`ed unlock. That lock is now gone, though the closure remained. It can now go too.